atlasdb-config/src/main/java/com/palantir/atlasdb/http/TextDelegateDecoder.java

```diff
@@ -59,15 +64,23 @@ public TextDelegateDecoder(Decoder delegate) {
     }
 
     @Override
-    public Object decode(Response response, Type type) throws IOException, DecodeException, FeignException {
-        Collection<String> contentTypes = response.headers().get(HttpHeaders.CONTENT_TYPE);
-        // In the case of multiple content types, or an unknown content type, we'll use the delegate instead.
-        if (contentTypes != null
-                && contentTypes.size() == 1
-                && Iterables.getOnlyElement(contentTypes, "").equals(MediaType.TEXT_PLAIN)) {
-            return stringDecoder.decode(response, type);
+    public Object decode(Response response, Type type) throws IOException, FeignException {
+        try {
+            Collection<String> contentTypes = response.headers().entrySet().stream()
+                    .filter(entry -> entry.getKey().compareToIgnoreCase(HttpHeaders.CONTENT_TYPE) == 0)
+                    .map(Map.Entry::getValue)
+                    .flatMap(Collection::stream)
+                    .collect(Collectors.toSet());
+            // In the case of multiple content types, or an unknown content type, we'll use the delegate instead.
+            if (contentTypes != null
+                    && contentTypes.size() == 1
+                    && Iterables.getOnlyElement(contentTypes, "").equals(MediaType.TEXT_PLAIN)) {
+                return stringDecoder.decode(response, type);
```

Comment:
I assume the root cause here is that this is picking up older version of Feign that does not yet have the fix for https://github.com/OpenFeign/feign/pull/418 . Note that http-remoting ran into this issue, see https://github.com/palantir/http-remoting/pull/156 and specifically its own [implementation of TextDelegateDecoder with case insensitive headers](https://github.com/palantir/http-remoting/pull/156/files#diff-300aea212cd0d3ee78463cf66522f820R45) that ideally we would use or at least keep in sync with.

`com.palantir.atlasdb.keyvalue.remoting.OctetStreamDelegateDecoder` also has same problem with case sensitive headers.

We can avoid doing the collect and instead stream everything:

```
            if (response.headers().entrySet().stream()
                    .filter(entry -> HttpHeaders.CONTENT_TYPE.equalsIgnoreCase(entry.getKey()))
                    .map(Map.Entry::getValue)
                    .flatMap(Collection::stream)
                    .anyMatch(MediaType.TEXT_PLAIN::equals)) {
                return stringDecoder.decode(response, type);
            }
```